certification/pyxis/pyxis.go

```diff
@@ -0,0 +1,179 @@
+package pyxis
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+)
+
+const (
+	apiVersion = "v1"
+)
+
+type HTTPClient interface {
+	Do(req *http.Request) (*http.Response, error)
+}
+
+type pyxisEngine struct {
+	ApiToken  string
+	ProjectId string
+	Client    HTTPClient
+}
+
+func getPyxisUrl(path string) string {
+	return fmt.Sprintf("https://%s/%s/%s", viper.GetString("pyxis-host"), apiVersion, path)
+}
+
+func NewPyxisEngine(apiToken string, projectId string, httpClient HTTPClient) *pyxisEngine {
+	return &pyxisEngine{
+		ApiToken:  apiToken,
+		ProjectId: projectId,
+		Client:    httpClient,
+	}
+}
+
+func (p *pyxisEngine) GetRedHatLayers() (*[]Layer, error) {
+	return &[]Layer{}, nil
+}
+
+func (p *pyxisEngine) createImage(ctx context.Context, certImage *CertImage) (*CertImage, error) {
+	b, err := json.Marshal(certImage)
+	if err != nil {
+		log.Error(err)
+		return nil, err
+	}
+	req, _ := newRequestWithApiToken(ctx, http.MethodPost, getPyxisUrl("/images"), p.ApiToken)
```

Comment:
It seems the `/` before `/images` is redundant.

Reply:
Good catch.